src/tree-manager/index.js

```diff
@@ -85,29 +85,37 @@ class TreeManager {
 
   /**
   * If the node didn't specify anything on its own
-  * figure out the initial state based on parent selections
+  * figure out the initial state for checked and disabled based on parent
   * @param {object} node [description]
   */
-  setInitialCheckState (node) {
-    if (node.checked === undefined) node.checked = this.getNodeCheckedState(node)
+  setInitialStatus (node) {
+    if (node.checked === undefined || node.disabled === undefined) {
+      const { parentCheckState, parentDisabledState } = this.getNodeStatus(node)
+      if (node.checked === undefined) node.checked = node.checked || parentCheckState
+      if (node.disabled === undefined) node.disabled = node.disabled || parentDisabledState
+    }
   }
 
   /**
-   * Figure out the check state based on parent selections.
+   * Figure out the checked and disabled state based on parent.
    * @param  {[type]} node    [description]
    * @param  {[type]} tree    [description]
    * @return {[type]}         [description]
    */
-  getNodeCheckedState (node) {
+  getNodeStatus (node) {
     let parentCheckState = false
+    let parentDisabledState = false
     let parent = node._parent
-    while (parent && !parentCheckState) {
+    while (parent && (!parentCheckState || !parentDisabledState)) {
```

Comment:
One prop may short-circuit the other. I don't think we can avoid multiple traversals (one per prop) here (with current traversal logic).

Reply:
Don't worry about this part for now. I think fixing the Cognitive Complexity will eliminate this problem as well.

Reply:
Nope, it's ok. It continues until there's no more parent or both (check and disabled) haven't been setted to true. So var are not supposed to short-circuit each other.